patches/server/1051-Properly-destroy-placed-blocks-on-the-end-platform.patch

```diff
@@ -0,0 +1,23 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: DerEchtePilz <81232921+DerEchtePilz@users.noreply.github.com>
+Date: Sun, 18 Aug 2024 13:05:31 +0200
+Subject: [PATCH] Properly destroy placed blocks on the end platform
+
+
+diff --git a/src/main/java/net/minecraft/world/level/levelgen/feature/EndPlatformFeature.java b/src/main/java/net/minecraft/world/level/levelgen/feature/EndPlatformFeature.java
+index 0bc659a8427b89b5e3211220c55b52eec6a20494..c04f940ddced59c546a7a8674c71ba74ff61272e 100644
+--- a/src/main/java/net/minecraft/world/level/levelgen/feature/EndPlatformFeature.java
++++ b/src/main/java/net/minecraft/world/level/levelgen/feature/EndPlatformFeature.java
+@@ -65,6 +65,12 @@ public class EndPlatformFeature extends Feature<NoneFeatureConfiguration> {
+ 
+         worldaccess.getLevel().getCraftServer().getPluginManager().callEvent(portalEvent);
+         if (!portalEvent.isCancelled()) {
++            // Paper start - Properly destroy placed blocks on the end platform
++            for (BlockState state : blockList.getList()) {
++                BlockPos pos = BlockPos.containing(state.getX(), state.getY(), state.getZ());
++                worldaccess.destroyBlock(pos, true);
++            }
```

Comment:
Make the loop variable type `CraftBlockState` and then you can use `CraftBlockState#getPosition` to directly get the BlockPos.

Also, this loop should be behind a check on that boolean flag that also guards the destroyBlock call from above.